integrations/consul/proxysql-consul

```diff
@@ -0,0 +1,211 @@
+    print 'Configs pushed successfully.'
+
+
+def build_multivalue_insert(table, rows):
+    # add quotes arround all values to make them strings in the sql query
+    quoted_values = [['"' + x + '"' for x in row] for row in rows]
+    # join each row in a values() group
+    row_join = [','.join(x) for x in quoted_values]
+    query = 'INSERT INTO %s VALUES(%s)' % (table, '),('.join(row_join))
+    return query
+
+def update_proxysql_config(table, rows):
+    admin_connection = MySQLdb.connect(config[CFG_PROXY_IFACE],
+            config[CFG_PROXY_USERNAME],
+            config[CFG_PROXY_PASSWORD],
+            port=config[CFG_PROXY_PORT],
+            db='main')
+    
+    # clear table
+    cursor = admin_connection.cursor()
+    cursor.execute('DELETE FROM %s' % table)
+    cursor.close()
+    
+    # insert values from Consul
+    insert_query = build_multivalue_insert(table, rows)
+    cursor = admin_connection.cursor()
+    cursor.execute(insert_query)
+    cursor.close()
+
+    # commit changes to runtine
+    load_query = LOAD_QUERY_BY_TABLE[table]
+    cursor = admin_connection.cursor()
+    cursor.execute(load_query)
+    cursor.close()
+
+    admin_connection.close()
+
+
+def update_config():
+    """
+    Reads all input from stdin that is passed by Consul and extracts the config
+    that was modified.
+
+    Consul is configured to watch all proxysql keys and returns all their values
+    when one key is modified. To determine the value that was actually modified
+    we use the CONSUL_INDEX env var written by Consul. This should match the
+    'ModifyIndex' field of the value that triggered the watch.
+
+    Values are returned as a JSON array. The actual value content is stored in
+    the 'Value' field, base64 encoded.
+    """
+    if 'CONSUL_INDEX' not in os.environ:
+        print 'Missing consul index on update request'
+        exit(1)
+    consul_index = int(os.environ['CONSUL_INDEX'])
+    consul_data = sys.stdin.read()
+
+    updated_value = None
+    values = json.loads(consul_data)
+    for value in values:
+        if 'ModifyIndex' in value and value['ModifyIndex'] == consul_index:
+           updated_value = value
+           break
+    if not updated_value:
+        print 'Failed to determine updated config from Consul data'
+        exit(1)
+
+    proxysql_config_json = base64.b64decode(updated_value['Value'])
+    proxysql_config = json.loads(proxysql_config_json)
+    table = proxysql_config['table']
+    rows = proxysql_config['rows']
+    _uuid = proxysql_config['uuid']
+    
+    if not rows:
+        print 'Empty config set for table: %s.' % table
+        return
+    if _uuid == config[CFG_UUID]:
+        print 'Ignoring self update.'
+        return
+    
+    update_proxysql_config(table, rows)
+    print 'Configs updated successfully.'
+
+if __name__ == '__main__':
+    read_config()
+    read_uuid()
+
+    if len(sys.argv) > 1:
+        mode = sys.argv[1]
```

Comment:
For parsing args, I'd recommend the [docopt](https://pypi.python.org/pypi/docopt) package.